pyccel/stdlib/STC_Extensions/Common_extensions.h

```diff
@@ -0,0 +1,31 @@
+#include <stdarg.h>
+
+static inline i_key c_JOIN(i_key,_min)(size_t count, ...) {
+    va_list args;
+    va_start(args, count);
+    i_key min_value = va_arg(args, i_key);
+    for (size_t i = 1; i < count; ++i) {
+        i_key value = va_arg(args, i_key);
+        if (value < min_value) {
+            min_value = value;
+        }
+    }
+    va_end(args);
+    return min_value;
+}
+
+static inline i_key c_JOIN(i_key, _max)(size_t count, ...) {
+    va_list args;
+    va_start(args, count);
+    i_key max_value = va_arg(args, i_key);
+    for (size_t i = 1; i < count; ++i) {
+        i_key value = va_arg(args, i_key);
+        if (value > max_value) {
+            max_value = value;
+        }
+    }
+    va_end(args);
+    return max_value;
+}
+
+#undef i_key
```

Comment:
You have `#undef i_key` in `Common_extensions.h` and `List_extensions.h`. Will this work if you want to use `pull_elem` and `min` in the same file?

Reply:
Yes, It works just fine. Normally, each header should undefine `i_key`.

Reply:
Yes there is no way to end up with this because including `<List_extensions.h>` automatically implies the following:
```C
#ifndef _TOOLS_LIST_VEC_INT64_T
#define _TOOLS_LIST_VEC_INT64_T
#define i_type vec_int64_t
#define i_key int64_t
#define i_use_cmp
#define i_more
#include <stc/vec.h>
#include <List_extensions.h>
#endif // _TOOLS_LIST_VEC_INT64_T
```
That's why including an extension header should invalidate the original header of the same `i_type`

Reply:
Ok sounds good :+1: